subgraphs/vesta-finance/src/entities/protocol.ts

```diff
@@ -0,0 +1,360 @@
+  financialsSnapshot.dailyTotalRevenueUSD =
+    financialsSnapshot.dailyTotalRevenueUSD.plus(revenueAmountUSD);
+  financialsSnapshot.save();
+
+  const market = getOrCreateMarket(asset);
+  market.cumulativeProtocolSideRevenueUSD =
+    market.cumulativeProtocolSideRevenueUSD.plus(revenueAmountUSD);
+  market.cumulativeTotalRevenueUSD =
+    market.cumulativeTotalRevenueUSD.plus(revenueAmountUSD);
+  market.save();
+
+  const marketDailySnapshot = getOrCreateMarketSnapshot(event, market);
+  marketDailySnapshot.dailyProtocolSideRevenueUSD =
+    marketDailySnapshot.dailyProtocolSideRevenueUSD.plus(revenueAmountUSD);
+  marketDailySnapshot.dailyTotalRevenueUSD =
+    marketDailySnapshot.dailyTotalRevenueUSD.plus(revenueAmountUSD);
+  marketDailySnapshot.save();
+
+  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, market);
+  marketHourlySnapshot.hourlyProtocolSideRevenueUSD =
+    marketHourlySnapshot.hourlyProtocolSideRevenueUSD.plus(revenueAmountUSD);
+  marketHourlySnapshot.hourlyTotalRevenueUSD =
+    marketHourlySnapshot.hourlyTotalRevenueUSD.plus(revenueAmountUSD);
+  marketHourlySnapshot.save();
+}
+
+export function addSupplySideRevenue(
+  event: ethereum.Event,
+  asset: Address,
+  revenueAmountUSD: BigDecimal
+): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.cumulativeSupplySideRevenueUSD =
+    protocol.cumulativeSupplySideRevenueUSD.plus(revenueAmountUSD);
+  protocol.cumulativeTotalRevenueUSD =
+    protocol.cumulativeTotalRevenueUSD.plus(revenueAmountUSD);
+  protocol.save();
+
+  const financialsSnapshot = getOrCreateFinancialsSnapshot(event, protocol);
+  financialsSnapshot.dailySupplySideRevenueUSD =
+    financialsSnapshot.dailySupplySideRevenueUSD.plus(revenueAmountUSD);
+  financialsSnapshot.dailyTotalRevenueUSD =
+    financialsSnapshot.dailyTotalRevenueUSD.plus(revenueAmountUSD);
+  financialsSnapshot.save();
+
+  const market = getOrCreateMarket(asset);
+  market.cumulativeSupplySideRevenueUSD =
+    market.cumulativeSupplySideRevenueUSD.plus(revenueAmountUSD);
+  market.cumulativeTotalRevenueUSD =
+    market.cumulativeTotalRevenueUSD.plus(revenueAmountUSD);
+  market.save();
+
+  const marketDailySnapshot = getOrCreateMarketSnapshot(event, market);
+  marketDailySnapshot.dailySupplySideRevenueUSD =
+    marketDailySnapshot.dailySupplySideRevenueUSD.plus(revenueAmountUSD);
+  marketDailySnapshot.dailyTotalRevenueUSD =
+    marketDailySnapshot.dailyTotalRevenueUSD.plus(revenueAmountUSD);
+  marketDailySnapshot.save();
+
+  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, market);
+  marketHourlySnapshot.hourlySupplySideRevenueUSD =
+    marketHourlySnapshot.hourlySupplySideRevenueUSD.plus(revenueAmountUSD);
+  marketHourlySnapshot.hourlyTotalRevenueUSD =
+    marketHourlySnapshot.hourlyTotalRevenueUSD.plus(revenueAmountUSD);
+  marketHourlySnapshot.save();
+}
+
+export function addProtocolBorrowVolume(
```

Comment:
same thing here, I am curious what the design choice was. I may be missing something

Reply:
Just removed these middle layer functions in commit at 4051ac8514ff8e59bd34e249b7a3342d8824c11b